internal/provider/egress_private_endpoint_resource.go

```diff
@@ -0,0 +1,554 @@
+		)
+		return
+	}
+
+	if cluster.CloudProvider == client.CLOUDPROVIDERTYPE_AZURE {
+		resp.Diagnostics.AddError(
+			"Error creating egress private endpoint",
+			"Egress private endpoints cannot be created on Azure clusters",
+		)
+		return
+	} else if cluster.Config.Dedicated == nil {
+		resp.Diagnostics.AddError(
+			"Error creating egress private endpoint",
+			"Egress private endpoints cannot be created on serverless clusters",
+		)
+		return
+	}
+
+	targetServiceType, err := client.NewEgressPrivateEndpointTargetServiceTypeTypeFromValue(plan.TargetServiceType.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating egress private endpoint",
+			fmt.Sprintf("Invalid target service type: %s", err),
+		)
+		return
+	}
+
+	// Retry create operation if cluster is locked
+	var createdEndpoint client.EgressPrivateEndpoint
+	if err := retry.RetryContext(ctx, egressPrivateEndpointCreateTimeout,
+		createEgressPrivateEndpoint(
+			ctx,
+			r.provider.service,
+			cluster.Id,
+			plan.Region.ValueString(),
+			plan.TargetServiceIdentifier.ValueString(),
+			*targetServiceType,
+			&createdEndpoint,
+		),
+	); err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating egress private endpoint",
+			fmt.Sprintf("Could not create egress private endpoint: %s", formatAPIErrorMessage(err)),
+		)
+		return
+	}
+
+	var endpoint client.EgressPrivateEndpoint
+	if err := retry.RetryContext(ctx, egressPrivateEndpointCreateTimeout,
+		waitForEgressEndpointCreatedFunc(
+			ctx,
+			r.provider.service,
+			cluster.Id,
+			createdEndpoint.Id,
+			&endpoint,
+		),
+	); err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating egress private endpoint",
+			fmt.Sprintf("Could not create egress private endpoint: %s", formatAPIErrorMessage(err)),
+		)
+		return
+	}
+
+	var state EgressPrivateEndpoint
+	loadEgressPrivateEndpointIntoTerraformState(&endpoint, cluster.Id, &state)
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+}
+
+// Delete implements resource.Resource.
+func (r *egressPrivateEndpointResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	if r.provider == nil || !r.provider.configured {
+		addConfigureProviderErr(&resp.Diagnostics)
+		return
+	}
+
+	var state EgressPrivateEndpoint
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if err := retry.RetryContext(ctx, egressPrivateEndpointDeleteTimeout,
+		deleteEgressEndpointFunc(
+			ctx,
+			r.provider.service,
+			state.ClusterID.ValueString(),
+			state.ID.ValueString(),
+		),
+	); err != nil {
+		resp.Diagnostics.AddError(
+			"Error deleting egress private endpoint",
+			fmt.Sprintf("Could not delete egress private endpoint: %s", formatAPIErrorMessage(err)),
+		)
+		return
+	}
+
+	if err := retry.RetryContext(ctx, egressPrivateEndpointDeleteTimeout,
+		waitForEgressEndpointDeletedFunc(
```

Comment:
same feedback as above.  It seems like it would be clearer to incorporate the wait into the delete function above.